core/modules/css/EmptyChunks.h

```diff
@@ -46,37 +46,52 @@ namespace lsst {
 namespace qserv {
 namespace css {
 
+class DbInterfaceMySql;
+
 /// High-level empty-chunk-tracking class. Tracks empty chunks
 /// per-database. In the future, we will likely migrate to a
 /// per-partitioning-group scheme, at which point, we will re-think
 /// the db-based dispatch as well (user tables in the partitioning
 /// group may be extremely sparse).
 class EmptyChunks {
```

Comment:
Or maybe even `EmptyChunks` class itself can be an abstract interface with different implementations. We should untangle this mess, I do not particularly like EmptyChunk interface too 🙂 Something for the future ticket, no need to clean all mess in one sweep.

Reply:
Empty chunks is just supposed to be a cache of empty chunk information read from elsewhere.

Reply:
What I do not like about it is that it knows about all possible ways in which empty chunk list can be stored, i.e. it has too many dependencies. In other words its not just a cache, it's also code that populates cache.

Reply:
I agree with Andy. In theory a better design would be to turn `EmptyChunk` into an interface (an abstract class) and to feed a shared pointer to the interface into the normal constructor of `CssAccess` as it used to be in the previous version of the code. This would also eliminate extra parameters passed into `CssAccess` which are presently forwarded to the multi-faced `ChunkList` (like database connector, file path, etc.). Then implement `EmptyChunk` in different ways, for specific sources of the empty chunks. In theory one can see the factory method to be added to class `EmptyChunk`, something along these lines:
```
class EmptyChunk: std::enable_shared_from_this<EmptyChunk>{
public:
    static std::shared_ptr< EmptyChunk> load(<db-connector>, <file-name>, etc...);

   /// The current interface goes below...
   ...
};
```